event/producer.go

```diff
@@ -0,0 +1,35 @@
+package event
+
+import (
+	"context"
+
+	dpkafka "github.com/ONSdigital/dp-kafka/v2"
+	"github.com/ONSdigital/dp-search-data-finder/models"
+	"github.com/ONSdigital/log.go/v2/log"
+)
+
+//go:generate moq -out ./mock/producer.go -pkg mock . Marshaller
+
+// Marshaller defines a type for marshalling the requested object into the required format.
+type Marshaller interface {
+	Marshal(s interface{}) ([]byte, error)
+}
+
+// ContentUpdatedProducer produces kafka messages for instances which have been successfully processed.
+type ContentUpdatedProducer struct {
+	Marshaller Marshaller
+	Producer   dpkafka.IProducer
+}
+
+// ContentUpdate produce a kafka message for an instance which has been successfully processed.
+func (p ContentUpdatedProducer) ContentUpdate(ctx context.Context, event models.ContentUpdated) error {
```

Comment:
I think it develop this needs to be behind the ```ContentUpdatedTopicFlag``` . Now I can see the flag being created but i don't see a switch for the producer to not publish message when this is turned off based on the flag ? Maybe i might have missed it.